### ISSUE TITLE
Indented multi-line arguments followed by indented chained function calls cause functions further down to be interpreted to end earlier than it should if those functions contain indented multi-line arguments

### ISSUE THREAD

**Issue by cheapsteak** [jashkenas/coffeescript#3866]
input:

``` coffeescript
gulp.task 'styles', ->
  styles = gulp
    .src paths.styles.source
    .pipe stylus
      'include css': true
    .on 'error', handleError

  styles = styles.pipe(CSSmin()) if production
```

desired output:

``` javascript
gulp.task('styles', function() {
  var styles;
  styles = gulp.src(paths.styles.source).pipe(sourcemaps.init()).pipe(stylus({
    'include css': true
  })).on('error', handleError);
  if (production) {
    return styles = styles.pipe(CSSmin());
  }
});
```

<hr>

Adding this at the top of the file causes the function for the gulp task to end early:
input:

``` coffeescript

obj = _({})
  .extend thingWithReallyLongName,
    thingWithReallyLongName,
    thingWithReallyLongName
  .value()

gulp.task 'styles', ->
  styles = gulp
    .src paths.styles.source
    .pipe stylus
      'include css': true
    .on 'error', handleError

  styles = styles.pipe(CSSmin()) if production
```

output:

``` javascript
var obj, styles;

obj = _({}).extend(thingWithReallyLongName, thingWithReallyLongName, thingWithReallyLongName).value();

gulp.task('styles', function() {
  var styles;
  return styles = gulp.src(paths.styles.source).pipe(stylus({
    'include css': true
  })).on('error', handleError);
});

if (production) {
  styles = styles.pipe(CSSmin());
}
```

The `if (production)` block should be inside of the function above it.  

<hr>

Wrapping parens around either of the indented arguments fixes it.  

e.g. 

``` cofeescript
.pipe stylus('include css': true)
```

OR 

``` coffeescript
obj = _({})
  .extend(thingWithReallyLongName,
    thingWithReallyLongName,
    thingWithReallyLongName)
  .value()
```

Both result in the desired output again.  

<hr>

It's also fixed if the last chained call after extend is not indented. 

``` coffeescript
obj = _({})
  .extend thingWithReallyLongName,
    thingWithReallyLongName,
    thingWithReallyLongName
.value()
```

Removing the indentation before `.value()` also results in desired output.

**Comment by xixixao** [comment]
This is most likely a dupe of #3484.

**Comment by GeoffreyBooth** [comment]
Looks like this is fixed in b28e398:

```js
gulp.task('styles', function() {
  var styles;
  styles = gulp.src(paths.styles.source).pipe(stylus({
    'include css': true
  })).on('error', handleError);
  if (production) {
    return styles = styles.pipe(CSSmin());
  }
});
```
